core_lib/src/structure/copymultiplekeyframesdialog.cpp

```diff
@@ -0,0 +1,403 @@
+
+
+    // SET text on labWarning and infolabels
+    ui->labWarning->setText("");
+    ui->labInfoAction->setText(getActiveTab().toUpper());
+    ui->labInfoToFromLayer->setText(tr("From: ") + ui->cBoxFromLayer->currentText() +
+                                    tr(" To: ") + ui->cBoxFromLayer->currentText());
+    // SET connections HEADER
+    connect(ui->sBoxFirstFrame, SIGNAL(valueChanged(int)), this, SLOT(setFirstFrame(int)));
+    connect(ui->sBoxLastFrame, SIGNAL(valueChanged(int)), this, SLOT(setLastFrame(int)));
+    connect(ui->cBoxFromLayer, SIGNAL(currentTextChanged(QString)), this, SLOT(setFromLayer(QString)));
+    connect(ui->tabWidget, SIGNAL(currentChanged(int)), this, SLOT(setMethodPicked(int)));
+    // SET connections COPY
+    connect(ui->sBoxNumLoops, SIGNAL(valueChanged(int)), this, SLOT(setNumLoops(int)));
+    connect(ui->sBoxStartFrame, SIGNAL(valueChanged(int)), this, SLOT(setStartFrame(int)));
+    connect(ui->cBoxCopyToLayer, SIGNAL(currentTextChanged(QString)), this, SLOT(setCopyToLayer(QString)));
+    // SET connections MOVE
+    connect(ui->sBoxMove, SIGNAL(valueChanged(int)), this, SLOT(setMoveStartFrame(int)));
+    connect(ui->cBoxMoveToLayer, SIGNAL(currentTextChanged(QString)), this, SLOT(setMoveToLayer(QString)));
+    // SET connections REVERSE
+    connect(ui->sBoxStartReverse, SIGNAL(valueChanged(int)), this, SLOT(setReverseFrom(int)));
+    // SET connections DELETE (none)
+
+    setStartEnd(0);
+    checkValidity();
+}
+
+int CopyMultiplekeyframesDialog::getFirstFrame()
+{
+    return ui->sBoxFirstFrame->value();
+}
+
+int CopyMultiplekeyframesDialog::getLastFrame()
+{
+    return ui->sBoxLastFrame->value();
+}
+
+int CopyMultiplekeyframesDialog::getManiStartAt()
+{
+    return mManiStartAt;
+}
+
+int CopyMultiplekeyframesDialog::getManiEndAt()
+{
+    return mManiEndAt;
+}
+
+QString CopyMultiplekeyframesDialog::getFromLayer()
+{
+    return ui->cBoxFromLayer->currentText();
+}
+
+QString CopyMultiplekeyframesDialog::getCopyToLayer()
+{
+    return ui->cBoxCopyToLayer->currentText();
+}
+
+QString CopyMultiplekeyframesDialog::getMoveToLayer()
+{
+    return ui->cBoxMoveToLayer->currentText();
+}
+
+QString CopyMultiplekeyframesDialog::getDeleteOnLayer()
+{
+    return ui->cBoxFromLayer->currentText();
+}
+
+int CopyMultiplekeyframesDialog::getNumLoops()
+{
+    return ui->sBoxNumLoops->value();
+}
+
+void CopyMultiplekeyframesDialog::setNumLoopsMax(int numLoopsMax)
+{
+    ui->sBoxNumLoops->setMaximum(numLoopsMax);
+}
+
+int CopyMultiplekeyframesDialog::getCopyStartFrame()
+{
+    return ui->sBoxStartFrame->value();
+}
+
+int CopyMultiplekeyframesDialog::getMoveStartFrame()
+{
+    return ui->sBoxMove->value();
+}
+
+int CopyMultiplekeyframesDialog::getReverseStartFrame()
+{
+    return ui->sBoxStartReverse->value();
+}
+
+QString CopyMultiplekeyframesDialog::getActiveTab()
+{
+    if (ui->tabWidget->currentIndex() == 0)
+    {
+        return "copy";
```

Comment:
This is not shown to the user. It is used in actioncommands.cpp to find out what action to perform.

Reply:
You might wanna check your own code again ;) Have a look at line 95…

Reply:
Oh yeah, I see. I forgot that.
Btw should this label be uppercase? I like it, but there has been like a no-capitol's movement.

Reply:
As I wrote somewhere else, caps are considered the equivalent of shouting, and of course you wouldn’t normally shout at other people unless they wouldn’t hear you otherwise (or you are angry at them, but let’s not get into that). When you just want to emphasize something, italics, boldface or sometimes underlining is usually the better option (or a combination of those, if it’s super extra important [but my former arts teacher would kill you for that]).